xpk.py

```diff
@@ -1763,6 +1829,43 @@ def check_if_workload_exists(args) -> bool:
   return False
 
 
+def check_if_workload_can_schedule(args, system):
+  """Check if workload can schedule based on the cluster resources (tpu_type and maximu VM in cluster).
+
+  Args:
+    args: user provided arguments for running the command.
+    system: system characteristics
+
+  Returns:
+    returns true if workload can schedule, otherwise returns false.
+  """
+  cluster_config_map = get_cluster_configmap(args)
+
+  # Prevents workload creation failure for existing clusters with no ConfigMap
+  if cluster_config_map is None:
```

Comment:
This is smart to allow it to be backward compatible with clusters created without this change. +1.